lib/puppet/file_system/windows.rb

```diff
@@ -134,10 +137,30 @@ def replace_file(path, mode = nil)
              dacl
            when 0640, 0600
              secure_dacl(current_sid)
```

Comment:
This will default to secure_dacl(current_sid, FULL_CONTROL, FULL_CONTROL), is this to permissive?

Reply:
It is but i don't want to change the existing functionality, just add the new modes. I'll fill a new ticket to proper solve this.

Reply:
`current_sid` is already granted `FULL_CONTROL` in https://github.com/puppetlabs/puppet/pull/7807/commits/a43cdf5ac35ae61bf38889052276e8971229a4b7#diff-d56421f468d3cfe2988bff9fb021f192R210

Also, the current owner of a file has the `WRITE_DAC` permission, so they can always change the DACL and grant themselves `FULL_CONTROL`.